(CYTOSCAPE-12532) Node label drag and move feature

jingjingbic opened the pull request.

jingjingbic committed:
Initial implementation of label drag and move feature.

jingjingbic committed:
Minor changes for CYTOSCAPE-12532. Using the updated cytoscape-3 icon
file. Node label selection Icon maps to ":" in the new font file.

jingjingbic committed:
Merge branch 'develop' into CYTOSCAPE-12532

# Conflicts:
#	ding-impl/ding-presentation-impl/src/main/java/org/cytoscape/ding/impl/InputHandlerGlassPane.java
#	ding-impl/ding-presentation-impl/src/main/java/org/cytoscape/ding/impl/NetworkPicker.java
#	swing-util-impl/src/main/resources/fonts/cytoscape-3.ttf

jingjingbic committed:
CYTOSCAPE-12532. Update the tooltip gif file for label selection button.

jingjingbic committed:
CYTOSCAPE-12532: Canceling the current label selection if mouse wheel or
keyboard event is triggered.

jingjingbic committed:
Merge branch 'develop' into CYTOSCAPE-12532

mikekucera reviewed (changes requested):
The changes look good, and the implementation works very well. I just have a few minor concerns…

- NetworkPicker.selectLabel() 
    - The nodes should come from snapshot.getSpacialIndex2D().queryOverlap(). Only labels for visible nodes are visible. Its unnecessary and inefficient to iterate over all nodes in the network when only a subset of them are visible.
    - Also the nodes returned by queryOverlap() are already sorted by z-order.
- NodeLabelChange
    - This class should not refer to the View<CyNode> object directly, just its SUID. This causes the following bug: move a label, then delete the node, then undo the delete, then undo the move, it doesn’t work. The node object needs to be looked up again using the SUID after undoing the delete.
- InputHandlerGlassPane
    - I don’t think it should select the node when you select the label. I think that dragging the label is a separate action from selecting or deselecting the nodes. I don’t think it should reset the user’s selection when they move a label. Also it causes the style and table panels to change their contents every time you click on a label which is distracting.

jingjingbic committed:
Changes to address Mike's comments.
1. Refactor NetworkPicker.selectLabel() function to use
snapshot.getSpaceiaIndex2D().queryOverlap().
2. Store SUID in NodeLabelChangeEdit function in case the node is
deleted and then restored.
